module/__init__.py

```diff
@@ -739,7 +765,38 @@ def __init__(self, unpacker):
 
 
 class Event(Response):
-    pass
+
+    def __init__(self, unpacker):
+        # This is here for debugging purposes!
+        self.unpacker = unpacker
+
+        Response.__init__(self, unpacker)
+
+        # If this is a xcb_ge_generic_event_t (response type 35) then we need a few more fields
+        if self.xge and isinstance(unpacker, CffiUnpacker):
+            self.extension, self.length, self.event_type, self.full_sequence = unpacker.unpack("xB2xIH22xI")
+
+            # There's some extra work to do if the event has data past the 32 byte boundary
+            if self.length:
+
+                # Calculate the size of the original buffer. This is 4 bytes short as it seems to omit the `full_sequence` field
+                buffer_size = 32 + (self.length * 4) + 4
+                
+                # Create a buffer object that's the full size
+                buffer = ffi.buffer(unpacker.cdata, buffer_size)
+
+                # Copy the event to the new buffer and skip the `full_sequence` field
+                # newbuffer[0:32] = buffer[0:32]
+                buffer[32:buffer_size - 5] = buffer[36: buffer_size - 1]
+
+                # Provide the resized buffer to the unpacker
+                unpacker.buf = buffer
+
+                # We need to prevent resizing as that will reset the buffer
+                # Resizing shouldn't be needed anyway as we have sized it correctly here
+                unpacker._resize = lambda *args, **kwargs: None
+
+            unpacker.offset = 6
```

Comment:
Here I guess this is `struct.calcsize` of the same thing you unpacked above; can you either use `struct.calcsize` or add a comment to that effect?